File: src/main/java/ru/zdoher/japs/changelog/Init_006_Kanji.java
package ru.zdoher.japs.changelog;

import com.github.cloudyrock.mongock.ChangeLog;
import com.github.cloudyrock.mongock.ChangeSet;
import org.springframework.data.mongodb.core.MongoTemplate;
import ru.zdoher.japs.domain.Kanji;
import ru.zdoher.japs.domain.Language;
import ru.zdoher.japs.domain.TranslateEntity;
import ru.zdoher.japs.domain.Word;

import java.util.List;
import java.util.Map;

@ChangeLog(order = "006")
public class Init_006_Kanji {
    private Map<String, Kanji> kanjiMap;
    private Map<String, Language> languageMap;
    private Map<String, Word> wordsMap;

    public Init_006_Kanji() {
        kanjiMap = InitMaps.kanjiMap;
        languageMap = InitMaps.languageMap;
        wordsMap = InitMaps.wordsMap;
    }

    @ChangeSet(order = "001", id = "initialKanji", author = "zdoh", runAlways = true)
    public void initKanji(MongoTemplate template) {
        kanjiMap.put("一", template.save(
                new Kanji("一",
                        List.of(new TranslateEntity(languageMap.get("ru"), "один"), new TranslateEntity(languageMap.get("en"), "one")),
                        List.of("イチ", "イツ"), List.of("ひと"), List.of(wordsMap.get("一つ")), wordsMap.get("一月"))));

        kanjiMap.put("二", template.save(
                new Kanji("二",
                        List.of(new TranslateEntity(languageMap.get("ru"), "два"), new TranslateEntity(languageMap.get("en"), "two")),
                        List.of("ニ"), List.of("ふた"), List.of(wordsMap.get("二つ")), wordsMap.get("二月"))));

        kanjiMap.put("三", template.save(
                new Kanji("三",
                        List.of(new TranslateEntity(languageMap.get("ru"), "три"), new TranslateEntity(languageMap.get("en"), "three")),
                        List.of("サン"), List.of("み", "みっ"), List.of(wordsMap.get("三つ")), wordsMap.get("三月"))));

        kanjiMap.put("四", template.save(
                new Kanji("四",
                        List.of(new TranslateEntity(languageMap.get("ru"), "четыре"), new TranslateEntity(languageMap.get("en"), "four")),
                        List.of("シ"), List.of("よ", "よつ", "よん"), List.of(wordsMap.get("四つ")), wordsMap.get("四月"))));

        kanjiMap.put("五", template.save(
                new Kanji("五",
                        List.of(new TranslateEntity(languageMap.get("ru"), "пять"), new TranslateEntity(languageMap.get("en"), "five")),
                        List.of("ゴ"), List.of("いつ"), List.of(wordsMap.get("五つ")), wordsMap.get("五月"))));

        kanjiMap.put("六", template.save(
                new Kanji("六",
                        List.of(new TranslateEntity(languageMap.get("ru"), "шесть"), new TranslateEntity(languageMap.get("en"), "six")),
                        List.of("ロク"), List.of("むつ", "むい"), List.of(wordsMap.get("六つ")), wordsMap.get("六月"))));

        kanjiMap.put("七", template.save(
                new Kanji("七",
                        List.of(new TranslateEntity(languageMap.get("ru"), "семь"), new TranslateEntity(languageMap.get("en"), "seven")),
                        List.of("シチ"), List.of("なな", "なの"), List.of(wordsMap.get("七つ")), wordsMap.get("七月"))));

        kanjiMap.put("八", template.save(
                new Kanji("八",
                        List.of(new TranslateEntity(languageMap.get("ru"), "восемь"), new TranslateEntity(languageMap.get("en"), "eight")),
                        List.of("ハチ"), List.of("や", "やつ", "よう"), List.of(wordsMap.get("八つ")), wordsMap.get("八月"))));

        kanjiMap.put("九", template.save(
                new Kanji("九",
                        List.of(new TranslateEntity(languageMap.get("ru"), "девять"), new TranslateEntity(languageMap.get("en"), "nine")),
                        List.of("キュウ", "ク"), List.of("ここの"), List.of(wordsMap.get("九つ")), wordsMap.get("九月"))));

        kanjiMap.put("十", template.save(
                new Kanji("十",
                        List.of(new TranslateEntity(languageMap.get("ru"), "пять"), new TranslateEntity(languageMap.get("en"), "five")),
                        List.of("サン"), List.of("と"), List.of(wordsMap.get("十")), wordsMap.get("十日"))));

        kanjiMap.put("人", template.save(
                new Kanji("人",
                        List.of(new TranslateEntity(languageMap.get("ru"), "человек")),
                        List.of("ジン", "ニン"), List.of(wordsMap.get("人")), wordsMap.get("日本人"))));

        kanjiMap.put("名", template.save(
                new Kanji("名",
                        List.of(new TranslateEntity(languageMap.get("ru"), "имя")),
                        List.of("メイ", "ミョウ"), List.of(wordsMap.get("名")), wordsMap.get("有名"))));

        kanjiMap.put("方", template.save(
                new Kanji("方",
                        List.of(new TranslateEntity(languageMap.get("ru"), "направление; сторона")),
                        List.of("ホウ"), List.of("かた"), null, wordsMap.get("方法"))));

        kanjiMap.put("本", template.save(
                new Kanji("本",
                        List.of(new TranslateEntity(languageMap.get("ru"), "книга; основа; источник")),
                        List.of("ホン"), List.of("もと"), null, wordsMap.get("日本"))));

        kanjiMap.put("日", template.save(
                new Kanji("日",
                        List.of(new TranslateEntity(languageMap.get("ru"), "день, солнце, счетчик для дней")),
                        List.of("ニチ", "ジツ"), List.of("か"), List.of(wordsMap.get("日")), wordsMap.get("毎日"))));

        kanjiMap.put("何", template.save(
                new Kanji("何",
                        List.of(new TranslateEntity(languageMap.get("ru"), "что")),
                        List.of("カ"), List.of(wordsMap.get("何")),  null)));

        kanjiMap.put("大", template.save(
                new Kanji("大",
                        List.of(new TranslateEntity(languageMap.get("ru"), "большой")),
                        List.of("ダイ", "タイ"), List.of("おお"), List.of(wordsMap.get("大きい"), wordsMap.get("大いに")), wordsMap.get("大学"))));

        kanjiMap.put("学", template.save(
                new Kanji("学",
                        List.of(new TranslateEntity(languageMap.get("ru"), "учиться; учеба; наука")),
                        List.of("ガク"), List.of(wordsMap.get("学ぶ")), wordsMap.get("大学"))));

        kanjiMap.put("社", template.save(
                new Kanji("社",
                        List.of(new TranslateEntity(languageMap.get("ru"), "фирма")),
                        List.of("シャ"), List.of(wordsMap.get("社")), wordsMap.get("社員"))));

        kanjiMap.put("員", template.save(
                new Kanji("員",
                        List.of(new TranslateEntity(languageMap.get("ru"), "работник; сотрудник; член персонала; суффикс профессии")),
                        List.of("イン"), null, null, wordsMap.get("社員"))));

        kanjiMap.put("会", template.save(
                new Kanji("会",
                        List.of(new TranslateEntity(languageMap.get("ru"), "встреча")),
                        List.of("カイ", "エ"), List.of(wordsMap.get("会う")), wordsMap.get("会社"))));

        kanjiMap.put("先", template.save(
                new Kanji("先",
                        List.of(new TranslateEntity(languageMap.get("ru"), "впереди; перед")),
                        List.of("セン"), List.of(wordsMap.get("先")), wordsMap.get("先生"))));

        kanjiMap.put("生", template.save(
                new Kanji("生",
                        List.of(new TranslateEntity(languageMap.get("ru"), "жизнь")),
                        List.of("セイ", "ショウ"),
                        List.of(wordsMap.get("生きる"), wordsMap.get("生かす"), wordsMap.get("生ける"), wordsMap.get("生まれる"),
                                wordsMap.get("生む"), wordsMap.get("生える"), wordsMap.get("生やす"), wordsMap.get("生")),
                        wordsMap.get("先生"))));

        kanjiMap.put("行", template.save(
                new Kanji("行",
                        List.of(new TranslateEntity(languageMap.get("ru"), "идти")),
                        List.of("コウ", "ギョウ", "アン"),  List.of(wordsMap.get("行く"), wordsMap.get("行う")), wordsMap.get("銀行"))));

        kanjiMap.put("来", template.save(
                new Kanji("来",
                        List.of(new TranslateEntity(languageMap.get("ru"), "приходить")),
                        List.of("ライ"),  List.of(wordsMap.get("来る")), wordsMap.get("来年"))));

        kanjiMap.put("車", template.save(
                new Kanji("車",
                        List.of(new TranslateEntity(languageMap.get("ru"), "машина; автомобиль")),
                        List.of("シャ"),  List.of(wordsMap.get("車")), wordsMap.get("電車"))));

        kanjiMap.put("月", template.save(
                new Kanji("月",
                        List.of(new TranslateEntity(languageMap.get("ru"), "месяц (календарный); месяц; луна")),
                        List.of("ゲツ", "ガツ"), List.of(wordsMap.get("月")), wordsMap.get("月曜日"))));

        kanjiMap.put("木", template.save(
                new Kanji("木",
                        List.of(new TranslateEntity(languageMap.get("ru"), "дерево")),
                        List.of("ボク", "モク"), List.of("こ"), List.of(wordsMap.get("木")), wordsMap.get("木曜日"))));

        kanjiMap.put("山", template.save(
                new Kanji("山",
                        List.of(new TranslateEntity(languageMap.get("ru"), "гора")),
                        List.of("サン"),  List.of(wordsMap.get("山")), wordsMap.get("火山"))));

        kanjiMap.put("川", template.save(
                new Kanji("川",
                        List.of(new TranslateEntity(languageMap.get("ru"), "река")),
                        List.of("セン"), List.of(wordsMap.get("川")), wordsMap.get("河川"))));

        kanjiMap.put("田", template.save(
                new Kanji("田",
                        List.of(new TranslateEntity(languageMap.get("ru"), "рисовое поле")),
                        List.of("デン"), List.of(wordsMap.get("田")), wordsMap.get("水田"))));

        kanjiMap.put("口", template.save(
                new Kanji("口",
                        List.of(new TranslateEntity(languageMap.get("ru"), "рот")),
                        List.of("コウ", "ク"), List.of(wordsMap.get("口")), wordsMap.get("人口"))));

        kanjiMap.put("門", template.save(
                new Kanji("門",
                        List.of(new TranslateEntity(languageMap.get("ru"), "ворота")),
                        List.of("モン"),List.of(wordsMap.get("門")), wordsMap.get("専門"))));

        kanjiMap.put("火", template.save(
                new Kanji("火",
                        List.of(new TranslateEntity(languageMap.get("ru"), "огонь")),
                        List.of("カ"), List.of("ほ"), List.of(wordsMap.get("火")), wordsMap.get("火曜日"))));

        kanjiMap.put("水", template.save(
                new Kanji("水",
                        List.of(new TranslateEntity(languageMap.get("ru"), "вода")),
                        List.of("スイ"), List.of(wordsMap.get("水")), wordsMap.get("水曜日"))));

        kanjiMap.put("金", template.save(
                new Kanji("金",
                        List.of(new TranslateEntity(languageMap.get("ru"), "золото; деньги; металл")),
                        List.of("キン", "コン"), List.of(wordsMap.get("金")), wordsMap.get("金曜日"))));

        kanjiMap.put("土", template.save(
                new Kanji("土",
                        List.of(new TranslateEntity(languageMap.get("ru"), "земля")),
                        List.of("ド", "ト"), List.of(wordsMap.get("土")), wordsMap.get("土曜日"))));

        kanjiMap.put("子", template.save(
                new Kanji("子",
                        List.of(new TranslateEntity(languageMap.get("ru"), "ребенок")),
                        List.of("シ", "ス"), List.of(wordsMap.get("子")), wordsMap.get("女子"))));

        kanjiMap.put("女", template.save(
                new Kanji("女",
                        List.of(new TranslateEntity(languageMap.get("ru"), "женщина")),
                        List.of("ジョ", "ニョ", "ニョウ"), List.of(wordsMap.get("女")), wordsMap.get("女子"))));

        kanjiMap.put("私", template.save(
                new Kanji("私",
                        List.of(new TranslateEntity(languageMap.get("ru"), "я; свой; личный; приватный")),
                        List.of("シ"), List.of(wordsMap.get("私")), wordsMap.get("私立"))));


/*
        kanjiList.put("", template.save(
                new Kanji("",
                        List.of(new TranslateEntity(languageMap.get("ru"), "")),
                        List.of(""), List.of(wordsMap.get("")), wordsMap.get(""))));
*/



    }

}

File: src/main/java/ru/zdoher/japs/changelog/Init_006_Kanji.kt
package ru.zdoher.japs.changelog

import com.github.cloudyrock.mongock.ChangeLog
import com.github.cloudyrock.mongock.ChangeSet
import org.springframework.data.mongodb.core.MongoTemplate
import ru.zdoher.japs.domain.Kanji
import ru.zdoher.japs.domain.Language
import ru.zdoher.japs.domain.TranslateEntity
import ru.zdoher.japs.domain.Word

@ChangeLog(order = "006")
class Init_006_Kanji {

    @ChangeSet(order = "001", id = "initialKanji", author = "zdoh", runAlways = true)
    fun initKanji(template: MongoTemplate) {
        InitMaps.kanjiMap["一"] = template.save(
            Kanji(
                kanji = "一",
                meaning = listOf(
                    TranslateEntity(InitMaps.languageMap["ru"], "один"),
                    TranslateEntity(InitMaps.languageMap["en"], "one")
                ),
                onyumi = listOf("イチ", "イツ"),
                kunyumi = listOf<String>("ひと"),
                words = listOf(InitMaps.wordsMap["一つ"]),
                anchor = InitMaps.wordsMap["一月"]!!
            )
        )
        InitMaps.kanjiMap["二"] = template.save(
            Kanji(
                kanji = "二",
                meaning = listOf(
                    TranslateEntity(InitMaps.languageMap["ru"], "два"),
                    TranslateEntity(InitMaps.languageMap["en"], "two")
                ),
                onyumi = listOf("ニ"),
                kunyumi = listOf<String>("ふた"),
                words = listOf(InitMaps.wordsMap["二つ"]),
                anchor = InitMaps.wordsMap["二月"]!!
            )
        )
        InitMaps.kanjiMap["三"] = template.save(
            Kanji(
                kanji = "三",
                meaning = listOf(
                    TranslateEntity(InitMaps.languageMap["ru"], "три"),
                    TranslateEntity(InitMaps.languageMap["en"], "three")
                ),
                onyumi = listOf<String>("サン"),
                kunyumi = listOf<String>("み", "みっ"),
                words = listOf(InitMaps.wordsMap["三つ"]),
                anchor = InitMaps.wordsMap["三月"]!!
            )
        )
        InitMaps.kanjiMap["四"] = template.save(
            Kanji(
                kanji = "四",
                meaning = listOf(
                    TranslateEntity(InitMaps.languageMap["ru"], "четыре"),
                    TranslateEntity(InitMaps.languageMap["en"], "four")
                ),
                onyumi = listOf<String>("シ"),
                kunyumi = listOf<String>("よ", "よつ", "よん"),
                words = listOf(InitMaps.wordsMap["四つ"]),
                anchor = InitMaps.wordsMap["四月"]!!
            )
        )
        InitMaps.kanjiMap["五"] = template.save(
            Kanji(
                kanji = "五",
                meaning = listOf(
                    TranslateEntity(InitMaps.languageMap["ru"], "пять"),
                    TranslateEntity(InitMaps.languageMap["en"], "five")
                ),
                onyumi = listOf<String>("ゴ"),
                kunyumi = listOf<String>("いつ"),
                words = listOf(InitMaps.wordsMap["五つ"]),
                anchor = InitMaps.wordsMap["五月"]!!
            )
        )
        InitMaps.kanjiMap["六"] = template.save(
            Kanji(
                kanji = "六",
                meaning = listOf(
                    TranslateEntity(InitMaps.languageMap["ru"], "шесть"),
                    TranslateEntity(InitMaps.languageMap["en"], "six")
                ),
                onyumi = listOf<String>("ロク"),
                kunyumi = listOf<String>("むつ", "むい"),
                words = listOf(InitMaps.wordsMap["六つ"]),
                anchor = InitMaps.wordsMap["六月"]!!
            )
        )
        InitMaps.kanjiMap["七"] = template.save(
            Kanji(
                kanji = "七",
                meaning = listOf(
                    TranslateEntity(InitMaps.languageMap["ru"], "семь"),
                    TranslateEntity(InitMaps.languageMap["en"], "seven")
                ),
                onyumi = listOf<String>("シチ"),
                kunyumi = listOf<String>("なな", "なの"),
                words = listOf(InitMaps.wordsMap["七つ"]),
                anchor =  InitMaps.wordsMap["七月"]!!
            )
        )
        InitMaps.kanjiMap["八"] = template.save(
            Kanji(
                kanji = "八",
                meaning = listOf(
                    TranslateEntity(InitMaps.languageMap["ru"], "восемь"),
                    TranslateEntity(InitMaps.languageMap["en"], "eight")
                ),
                onyumi = listOf<String>("ハチ"),
                kunyumi = listOf<String>("や", "やつ", "よう"),
                words = listOf(InitMaps.wordsMap["八つ"]),
                anchor = InitMaps.wordsMap["八月"]!!
            )
        )
        InitMaps.kanjiMap["九"] = template.save(
            Kanji(
                kanji = "九",
                meaning = listOf(
                    TranslateEntity(InitMaps.languageMap["ru"], "девять"),
                    TranslateEntity(InitMaps.languageMap["en"], "nine")
                ),
                onyumi = listOf<String>("キュウ", "ク"),
                kunyumi = listOf<String>("ここの"),
                words = listOf(InitMaps.wordsMap["九つ"]),
                anchor = InitMaps.wordsMap["九月"]!!
            )
        )
        InitMaps.kanjiMap["十"] = template.save(
            Kanji(
                kanji = "十",
                meaning = listOf(
                    TranslateEntity(InitMaps.languageMap["ru"], "пять"),
                    TranslateEntity(InitMaps.languageMap["en"], "five")
                ),
                onyumi = listOf<String>("サン"),
                kunyumi = listOf<String>("と"),
                words = listOf(InitMaps.wordsMap["十"]),
                anchor = InitMaps.wordsMap["十日"]!!
            )
        )
        InitMaps.kanjiMap["人"] = template.save(
            Kanji(
                kanji = "人",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "человек")),
                onyumi = listOf<String>("ジン", "ニン"),
                words = listOf(InitMaps.wordsMap["人"]),
                anchor = InitMaps.wordsMap["日本人"]!!
            )
        )
        InitMaps.kanjiMap["名"] = template.save(
            Kanji(
                kanji = "名",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "имя")),
                onyumi = listOf<String>("メイ", "ミョウ"),
                words = listOf(InitMaps.wordsMap["名"]),
                anchor = InitMaps.wordsMap["有名"]!!
            )
        )
        InitMaps.kanjiMap["方"] = template.save(
            Kanji(
                kanji = "方",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "направление; сторона")),
                onyumi = listOf<String>("ホウ"),
                kunyumi =  listOf<String>("かた"),
                anchor = InitMaps.wordsMap["方法"]!!
            )
        )
        InitMaps.kanjiMap["本"] = template.save(
            Kanji(
                kanji = "本",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "книга; основа; источник")),
                onyumi = listOf<String>("ホン"),
                kunyumi = listOf<String>("もと"),
                anchor = InitMaps.wordsMap["日本"]!!
            )
        )
        InitMaps.kanjiMap["日"] = template.save(
            Kanji(
                kanji = "日",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "день, солнце, счетчик для дней")),
                onyumi = listOf<String>("ニチ", "ジツ"),
                kunyumi = listOf<String>("か"),
                words = listOf(InitMaps.wordsMap["日"]),
                anchor = InitMaps.wordsMap["毎日"]!!
            )
        )
        InitMaps.kanjiMap["何"] = template.save(
            Kanji(
                kanji = "何",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "что")),
                onyumi = listOf<String>("カ"),
                words = listOf(InitMaps.wordsMap["何"]),
                anchor = InitMaps.wordsMap["何"]!!
            )
        )
        InitMaps.kanjiMap["大"] = template.save(
            Kanji(
                kanji = "大",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "большой")),
                onyumi = listOf<String>("ダイ", "タイ"),
                kunyumi = listOf<String>("おお"),
                words = listOf(InitMaps.wordsMap["大きい"], InitMaps.wordsMap["大いに"]),
                anchor = InitMaps.wordsMap["大学"]!!
            )
        )
        InitMaps.kanjiMap["学"] = template.save(
            Kanji(
                kanji = "学",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "учиться; учеба; наука")),
                onyumi = listOf<String>("ガク"),
                words = listOf(InitMaps.wordsMap["学ぶ"]),
                anchor = InitMaps.wordsMap["大学"]!!
            )
        )
        InitMaps.kanjiMap["社"] = template.save(
            Kanji(
                kanji = "社",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "фирма")),
                onyumi = listOf<String>("シャ"),
                words = listOf(InitMaps.wordsMap["社"]),
                anchor = InitMaps.wordsMap["社員"]!!
            )
        )
        InitMaps.kanjiMap["員"] = template.save(
            Kanji(
                kanji = "員",
                meaning = listOf(
                    TranslateEntity(InitMaps.languageMap["ru"], "работник; сотрудник; член персонала; суффикс профессии")
                ),
                onyumi = listOf<String>("イン"),
                anchor = InitMaps.wordsMap["社員"]!!
            )
        )
        InitMaps.kanjiMap["会"] = template.save(
            Kanji(
                kanji = "会",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "встреча")),
                onyumi = listOf<String>("カイ", "エ"),
                words = listOf(InitMaps.wordsMap["会う"]),
                anchor = InitMaps.wordsMap["会社"]!!
            )
        )
        InitMaps.kanjiMap["先"] = template.save(
            Kanji(
                kanji = "先",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "впереди; перед")),
                onyumi = listOf<String>("セン"),
                words = listOf(InitMaps.wordsMap["先"]),
                anchor = InitMaps.wordsMap["先生"]!!
            )
        )
        InitMaps.kanjiMap["生"] = template.save(
            Kanji(
                kanji = "生",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "жизнь")),
                onyumi = listOf<String>("セイ", "ショウ"),
                words = listOf(
                    InitMaps.wordsMap["生きる"],
                    InitMaps.wordsMap["生かす"],
                    InitMaps.wordsMap["生ける"],
                    InitMaps.wordsMap["生まれる"],
                    InitMaps.wordsMap["生む"],
                    InitMaps.wordsMap["生える"],
                    InitMaps.wordsMap["生やす"],
                    InitMaps.wordsMap["生"]
                ),
                anchor = InitMaps.wordsMap["先生"]!!
            )
        )
        InitMaps.kanjiMap["行"] = template.save(
            Kanji(
                kanji = "行",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "идти")),
                onyumi = listOf<String>("コウ", "ギョウ", "アン"),
                words = listOf(InitMaps.wordsMap["行く"], InitMaps.wordsMap["行う"]),
                anchor = InitMaps.wordsMap["銀行"]!!
            )
        )
        InitMaps.kanjiMap["来"] = template.save(
            Kanji(
                kanji = "来",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "приходить")),
                onyumi = listOf<String>("ライ"),
                words = listOf(InitMaps.wordsMap["来る"]),
                anchor = InitMaps.wordsMap["来年"]!!
            )
        )
        InitMaps.kanjiMap["車"] = template.save(
            Kanji(
                kanji = "車",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "машина; автомобиль")),
                onyumi = listOf<String>("シャ"),
                words = listOf(InitMaps.wordsMap["車"]),
                anchor = InitMaps.wordsMap["電車"]!!
            )
        )
        InitMaps.kanjiMap["月"] = template.save(
            Kanji(
                kanji = "月",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "месяц (календарный); месяц; луна")),
                onyumi = listOf<String>("ゲツ", "ガツ"),
                words = listOf(InitMaps.wordsMap["月"]),
                anchor = InitMaps.wordsMap["月曜日"]!!
            )
        )
        InitMaps.kanjiMap["木"] = template.save(
            Kanji(
                kanji = "木",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "дерево")),
                onyumi = listOf<String>("ボク", "モク"),
                kunyumi = listOf<String>("こ"),
                words = listOf(InitMaps.wordsMap["木"]),
                anchor = InitMaps.wordsMap["木曜日"]!!
            )
        )
        InitMaps.kanjiMap["山"] = template.save(
            Kanji(
                kanji = "山",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "гора")),
                onyumi = listOf<String>("サン"),
                words = listOf(InitMaps.wordsMap["山"]),
                anchor = InitMaps.wordsMap["火山"]!!
            )
        )
        InitMaps.kanjiMap["川"] = template.save(
            Kanji(
                kanji = "川",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "река")),
                onyumi = listOf<String>("セン"),
                words = listOf(InitMaps.wordsMap["川"]),
                anchor = InitMaps.wordsMap["河川"]!!
            )
        )
        InitMaps.kanjiMap["田"] = template.save(
            Kanji(
                kanji = "田",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "рисовое поле")),
                onyumi = listOf<String>("デン"),
                words = listOf(InitMaps.wordsMap["田"]),
                anchor = InitMaps.wordsMap["水田"]!!
            )
        )
        InitMaps.kanjiMap["口"] = template.save(
            Kanji(
                kanji = "口",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "рот")),
                onyumi = listOf<String>("コウ", "ク"),
                words = listOf(InitMaps.wordsMap["口"]),
                anchor = InitMaps.wordsMap["人口"]!!
            )
        )
        InitMaps.kanjiMap["門"] = template.save(
            Kanji(
                kanji = "門",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "ворота")),
                onyumi = listOf<String>("モン"),
                words = listOf(InitMaps.wordsMap["門"]),
                anchor = InitMaps.wordsMap["専門"]!!
            )
        )
        InitMaps.kanjiMap["火"] = template.save(
            Kanji(
                kanji = "火",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "огонь")),
                onyumi = listOf<String>("カ"),
                kunyumi = listOf<String>("ほ"),
                words = listOf(InitMaps.wordsMap["火"]),
                anchor = InitMaps.wordsMap["火曜日"]!!
            )
        )
        InitMaps.kanjiMap["水"] = template.save(
            Kanji(
                kanji = "水",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "вода")),
                onyumi = listOf<String>("スイ"),
                words = listOf(InitMaps.wordsMap["水"]),
                anchor = InitMaps.wordsMap["水曜日"]!!
            )
        )
        InitMaps.kanjiMap["金"] = template.save(
            Kanji(
                kanji = "金",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "золото; деньги; металл")),
                onyumi = listOf<String>("キン", "コン"),
                words = listOf(InitMaps.wordsMap["金"]),
                anchor = InitMaps.wordsMap["金曜日"]!!
            )
        )
        InitMaps.kanjiMap["土"] = template.save(
            Kanji(
                kanji = "土",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "земля")),
                onyumi = listOf<String>("ド", "ト"),
                words = listOf(InitMaps.wordsMap["土"]),
                anchor = InitMaps.wordsMap["土曜日"]!!
            )
        )
        InitMaps.kanjiMap["子"] = template.save(
            Kanji(
                kanji = "子",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "ребенок")),
                onyumi = listOf<String>("シ", "ス"),
                words = listOf(InitMaps.wordsMap["子"]),
                anchor = InitMaps.wordsMap["女子"]!!
            )
        )
        InitMaps.kanjiMap["女"] = template.save(
            Kanji(
                kanji = "女",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "женщина")),
                onyumi = listOf<String>("ジョ", "ニョ", "ニョウ"),
                words = listOf(InitMaps.wordsMap["女"]),
                anchor = InitMaps.wordsMap["女子"]!!
            )
        )
        InitMaps.kanjiMap["私"] = template.save(
            Kanji(
                kanji = "私",
                meaning = listOf(TranslateEntity(InitMaps.languageMap["ru"], "я; свой; личный; приватный")),
                onyumi = listOf<String>("シ"),
                words = listOf(InitMaps.wordsMap["私"]),
                anchor = InitMaps.wordsMap["私立"]!!
            )
        )
/*
        kanjiList.put("", template.save(
                new Kanji("",
                        listOf(new TranslateEntity(InitMaps.languageMap.get("ru"), "")),
                        listOf(""), listOf(InitMaps.wordsMap.get("")), InitMaps.wordsMap.get(""))));
*/
    }
}